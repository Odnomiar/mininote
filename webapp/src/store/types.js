const state = {
  SELECTED_NOTE_ID: 'selectedNoteId',
  NOTEBOOKS: 'notebooks',
  VERSION: 'version',
}
const getters = {
  LOADED: 'loaded',
  DIRTY: 'dirty',
  NOTES: 'notes',
  SELECTED_NOTE: 'selectedNote',
  NOTE_BY_ID: 'noteById',
  CURRENT_CONTENT: 'currentContent',
  CURRENT_TITLE: 'currentTitle',
}
const actions = {
  LOAD_NOTEBOOK: 'loadNotebook',
  LIST_NOTEBOOKS: 'listNotebooks',
  CREATE_NOTEBOOK: 'createNotebook',
  UPDATE_NOTEBOOK: 'updateNotebook',
  UPDATE_NOTE: 'updateNote',
  APPLY_CHANGES: 'applyChanges',
}
const mutations = {
  SELECT_NOTE: 'selectNote',
  SET_LOAD_NOTEBOOK: 'setLoadNotebook',
  SELECT_FIRST: 'selectFirst',
  ADD_CHANGE: 'addChange',
  REVERT_CHANGES: 'revertChanges',
  RESET: 'reset',
}

export { state, getters, actions, mutations }
