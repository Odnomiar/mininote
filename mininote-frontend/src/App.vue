<template>
  <div id="app">
    <div class="logo">
      <span>✎ Mini_Note</span>
      <span class="small">BETA </span>
    </div>
    <div class="container-fluid">
      <b-alert show v-if="alert && alert.variant === 'danger'" variant="danger">{{ alert.text }}</b-alert>
      <b-alert show v-if="alert && alert.variant === 'success'" variant="success">{{ alert.text }}</b-alert>
      <control-bar @alert="showAlert" :hasChanges="hasChanges"></control-bar>
      <div class="row" v-if="loaded">
        <div class="col-2">
          <notes-picker @noteSelected="onNoteSelected" @alert="showAlert" @addNote="addNote" @deleteNote="deleteNote"></notes-picker>
        </div>
        <div class="col-10" v-if="selectedNoteId >= 0">
          <notes-editor :content="currentContent" :id="selectedNoteId" :dirty="dirty" @alert="showAlert" @contentUpdate="onNoteUpdated"></notes-editor>
        </div>
      </div>
      <div v-if="!loaded">
        <div class="placeholder">
          <span>Please open an existing notebook or create a new one.</span>
        </div>
      </div>
    </div>
    <div class="footer">Made w/ &#x2661; by
      <a href="http://ferdinand-muetsch.de">Ferdinand Mütsch</a> |
      <a href="http://github.com/n1try/mininote">GitHub</a>
    </div>
  </div>
</template>

<script>
import { mapState, mapGetters } from 'vuex'

import NotesEditor from './components/NotesEditor'
import NotesPicker from './components/NotesPicker'
import ControlBar from './components/ControlBar'

export default {
  name: 'app',
  data() {
    return {
      alert: null
    }
  },
  computed: {
    ...mapState([
      'selectedNoteId'
    ]),
    ...mapGetters([
      'loaded',
      'selectedNote',
      'dirty',
      'currentContent',
      'noteById'
    ]),
    hasChanges: function() {
      return false
    }
  },
  components: {
    NotesEditor,
    NotesPicker,
    ControlBar
  },
  created() {
    window.addEventListener("beforeunload", this.didIntentToLeave);
  },
  methods: {
    onNoteSelected: function(noteId) {
      this.$store.commit('selectNote', noteId)
    },
    onNotesLoaded: function(notes) {
    },
    addNote: function(note) {
      this.notes.push(note)
    },
    deleteNote: function(note) {
      this.notes.splice(this.notes.indexOf(note), 1)
    },
    discardChanges: function() {
      this.notes = JSON.parse(JSON.stringify(this.notesInitial))
    },
    showAlert: function(text, variant) {
      let vm = this;
      if (!variant) variant = 'danger'
      vm.alert = { text, variant }
      setTimeout(function() {
        vm.alert = null
      }, 3000)
    },
    onNoteUpdated: function({content, id}) {
      this.$store.commit('addChange', { type: 'update', payload: { ...this.noteById(id), content }})
    },
    didIntentToLeave: function(event) {
      if(this.hasChanges){
        // Cancel the event as stated by the standard.
        event.preventDefault();
        // Chrome requires returnValue to be set.
        event.returnValue = '';
      }
    }
  }
}
</script>

<style>
#app {
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #2c3e50;
  position: absolute;
  width: 100%;
}

#app a {
  color: #42B983;
  text-decoration: none;
}

#app .logo {
  color: #bbb;
  font-size: 24px;
  position: absolute;
  left: 20px;
}

#app .logo .small {
  color: #42B983;
  font-size: 12px;
}

#app .footer {
  color: #bbb;
  position: fixed;
  bottom: 10px;
  left: 20px;
  font-size: 12px;
}

button {
  cursor: pointer;
}

#app .btn-primary {
  background-color: #42B983;
  border-color: #42B983;
}

#app .placeholder {
  font-size: 40px;
  text-align: center;
  color: rgb(73, 80, 87);
  position: absolute;
  width: 600px;
  top: 35vh;
  left: calc(50% - 300px);
}

.alert {
  width: 500px;
  position: fixed;
  top: 15px;
  left: calc(50% - 250px);
  z-index: 999;
  text-align: center;
}
</style>
