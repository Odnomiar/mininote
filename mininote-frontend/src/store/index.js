import Vue from 'vue'
import Vuex from 'vuex'

import api from '../api'

Vue.use(Vuex)

const emptyState = {
    notebook: {},
    notebooks: [],
    changes: {
        add: [],
        delete: [],
        update: {}
    },
    selectedNoteId: -1,
    loadNotebookId: '',
    version: require('../../package.json').version
}

/// GETTERS ///
function loaded(state) {
    return state.notebook.hasOwnProperty('notes')
}

function dirty(state) {
    return !!(state.changes.add.length || state.changes.delete.length || Object.keys(state.changes.update).length)
}

function selectedNote(state) {
    return resolveNotes(state).find(n => n.id === state.selectedNoteId)
}

function noteById(state) {
    return (id) => resolveNotes(state).find(n => n.id === id)
}

function resolveNotes(state) {
    let notes = []
    notes = notes.concat(state.notebook.notes || [])
    notes = notes.concat(state.changes.add)
    notes = notes.filter(n1 => !state.changes.delete.some(n2 => n1.id === n2.id))
    return notes
}

function currentContent(state) {
    let id = state.selectedNoteId
    if (id < 0) return ''
    if (state.changes.update.hasOwnProperty(id)) return state.changes.update[id].content
    return selectedNote(state).content
}

function currentTitle(state) {
    let id = state.selectedNoteId
    if (id < 0) return ''
    if (state.changes.update.hasOwnProperty(id)) return state.changes.update[id].title
    return selectedNote(state).title
}

/// MUTATIONS ///
function selectNote(state, id) {
    if (id < 0) return null
    let notes = resolveNotes(state)
    if (notes.some(n => n.id === id)) {
        Vue.set(state, 'selectedNoteId', id)
    }
}

function selectFirst(state) {
    let notes = resolveNotes(state)
    let id = notes.length ? notes[0].id : -1
    Vue.set(state, 'selectedNoteId', id)
}

/// STORE ///
const store = new Vuex.Store({
    state: JSON.parse(JSON.stringify(emptyState)),
    getters: {
        selectedNote,
        dirty,
        notes: resolveNotes,
        currentContent,
        currentTitle,
        loaded,
        noteById
    },
    mutations: {
        setNotebooks(state, notebookList) {
            Vue.set(state, 'notebooks', notebookList)
        },
        setNotebook(state, { id, password }) {
            Vue.set(state.notebook, 'id', id)
            Vue.set(state.notebook, 'password', password)
        },
        setNotes(state, notes) {
            Vue.set(state.notebook, 'notes', notes)
        },
        addNote(state, note) {
            Vue.set(state.notebook, 'notes', state.notebook.notes.concat([note]))
            if (state.selectedNoteId === note.oldId) selectNote(state, note.id)
        },
        removeNote(state, note) {
            Vue.set(state.notebook, 'notes', state.notebook.notes.filter(n => n.id !== note.id))
            if (state.selectedNoteId === note.id && state.notebook.notes.length) {
                Vue.set(state, 'selectedNoteId', state.notes[0].id)
            }
        },
        updateNote(state, note) {
            let notes = resolveNotes(state)
            let found = notes.find(n => n.id === note.id)
            for (let prop in found) {
                Vue.set(found, prop, note[prop])
            }
        },
        selectNote,
        selectFirst,
        setLoadNotebook(state, id) {
            Vue.set(state, 'loadNotebookId', id)
        },
        addChange(state, { type, payload }) {
            switch (type) {
                case 'add':
                case 'delete':
                    Vue.set(state.changes, type, state.changes[type].concat([payload]))
                    break
                case 'update':
                    let hasChange = state.changes.update.hasOwnProperty(payload.id)
                    let originalNote = (state.notebook.notes || []).find(n => n.id === payload.id)

                    if (originalNote && originalNote.content === payload.content && hasChange) {
                        Vue.delete(state.changes.update, payload.id)
                    } else {
                        Vue.set(state.changes.update, payload.id, payload)
                    }
                    break
            }
        },
        removeChange(state, { type, payload }) {
            switch (type) {
                case 'add':
                case 'delete':
                    let idx = state.changes[type].findIndex(n => n.id === payload.id)
                    Vue.delete(state.changes[type], idx)
                    break
                case 'update':
                    Vue.delete(state.changes.update, payload.id)
            }
        },
        revertChanges(state) {
            Vue.set(state.changes, 'add', [])
            Vue.set(state.changes, 'delete', [])

            Object.keys(state.changes.update).forEach(id => {
                let note = (state.notebook.notes || []).find(n => n.id === parseInt(id))
                let content = note.content

                Vue.set(note, 'content', '')
                Vue.set(note, 'content', content)
            })

            Vue.set(state.changes, 'update', {})
        },
        reset(state) {
            for (let prop in emptyState) {
                Vue.set(state, prop, JSON.parse(JSON.stringify(emptyState[prop])))
            }
        }
    },
    actions: {
        listNotebooks({ commit }) {
            return api.list()
                .then(data => commit('setNotebooks', data))
        },
        loadNotebook({ commit }, { id, password }) {
            return api.getNotes(id, password)
                .then(data => commit('setNotes', data))
                .then(() => commit('setNotebook', { id, password }))
        },
        createNotebook({ commit }, { id, password }) {
            return api.create(id, password)
                .then(data => commit('setNotes', data.notes))
                .then(() => commit('setNotebook', { id, password }))
        },
        updateNotebook({ state, commit }, notebook) {
            let updateNotebook = JSON.parse(JSON.stringify(notebook))
            if (notebook.id === state.notebook.id) delete updateNotebook.id
            return api.update(state.notebook.id, state.notebook.password, updateNotebook)
                .then(() => commit('setNotebook', notebook))
        },
        createNote({ state, commit }, note) {
            return api.addNote(state.notebook.id, state.notebook.password, note)
                .then(data => commit('addNote', { ...note, id: data.id, oldId: note.id }))
        },
        updateNote({ state, commit }, note) {
            return api.updateNote(state.notebook.id, state.notebook.password, note)
                .then(() => commit('updateNote', note))
        },
        deleteNote({ state, commit }, note) {
            return api.deleteNote(state.notebook.id, state.notebook.password, note)
                .then(() => commit('removeNote', note))
        },
        applyChanges({ state, commit, dispatch }) {
            let promises = []

            Object.values(state.changes.update).forEach(n => {
                let newNote = state.changes.add.find(n1 => n1.id === n.id)
                let deletedNote = state.changes.delete.find(n1 => n1.id === n.id)

                if (newNote) {
                    commit('updateNote', n)
                    commit('removeChange', { type: 'update', payload: n })
                }
                else if (deletedNote) commit('removeChange', { type: 'update', payload: n })
                else promises.push(dispatch('updateNote', n).then(() => commit('removeChange', { type: 'update', payload: n })))
            })

            state.changes.add.forEach(n => {
                if (state.changes.delete.some(n1 => n1.id === n.id)) {
                    commit('removeChange', { type: 'add', payload: n })
                    commit('removeChange', { type: 'delete', payload: n })
                }
                else promises.push(dispatch('createNote', n).then(() => commit('removeChange', { type: 'add', payload: n })))
            })
            state.changes.delete.forEach(n => {
                promises.push(dispatch('deleteNote', n).then(() => commit('removeChange', { type: 'delete', payload: n })))
            })

            return Promise.all(promises)
        }
    },
    strict: true
})

export default store