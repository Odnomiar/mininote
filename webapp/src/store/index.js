import Vuex from 'vuex'

import api from '../api'

import { version } from '../../package.json'

/* eslint-disable no-case-declarations */

const emptyState = {
  notebook: {},
  notebooks: [],
  changes: {
    add: [],
    delete: [],
    update: {},
  },
  selectedNoteId: -1,
  loadNotebookId: '',
  version,
}

/// GETTERS ///
function loaded(state) {
  return !!state.notebook.notes
}

function dirty(state) {
  return !!(
    state.changes.add.length ||
    state.changes.delete.length ||
    Object.keys(state.changes.update).length
  )
}

function selectedNote(state) {
  return resolveNotes(state).find((n) => n.id === state.selectedNoteId)
}

function noteById(state) {
  return (id) => resolveNotes(state).find((n) => n.id === id)
}

function resolveNotes(state) {
  let notes = []
  notes = notes.concat(state.notebook.notes || [])
  notes = notes.concat(state.changes.add)
  notes = notes.filter(
    (n1) => !state.changes.delete.some((n2) => n1.id === n2.id),
  )
  return notes
}

function currentContent(state) {
  let id = state.selectedNoteId
  if (id < 0) return ''
  if (id in state.changes.update) return state.changes.update[id].content
  const note = selectedNote(state)
  return note ? note.content : ''
}

function currentTitle(state) {
  let id = state.selectedNoteId
  if (id < 0) return ''
  if (id in state.changes.update) return state.changes.update[id].title
  const note = selectedNote(state)
  return note ? note.title : ''
}

/// MUTATIONS ///
function selectNote(state, id) {
  if (id < 0) state.selectedNoteId = -1
  let notes = resolveNotes(state)
  if (notes.some((n) => n.id === id)) {
    state.selectedNoteId = id
  }
}

function selectFirst(state) {
  let notes = resolveNotes(state)
  let id = notes.length ? notes[0].id : -1
  state.selectedNoteId = id
}

function updateUrl(state) {
  location.hash = state.notebook.id ? `#/notebook/${state.notebook.id}` : ''
}

/// STORE ///
const store = new Vuex.Store({
  state: JSON.parse(JSON.stringify(emptyState)),
  getters: {
    selectedNote,
    dirty,
    notes: resolveNotes,
    currentContent,
    currentTitle,
    loaded,
    noteById,
  },
  mutations: {
    setNotebooks(state, notebookList) {
      state.notebooks = notebookList
    },
    setNotebook(state, { id, password }) {
      state.notebook.id = id
      state.notebook.password = password
      updateUrl(state)
    },
    setNotes(state, notes) {
      state.notebook.notes = notes
    },
    addNote(state, note) {
      state.notebook.notes = state.notebook.notes.concat([note])
      if (state.selectedNoteId === note.oldId) selectNote(state, note.id)
    },
    removeNote(state, note) {
      state.notebook.notes = state.notebook.notes.filter(
        (n) => n.id !== note.id,
      )
      if (state.selectedNoteId === note.id && state.notebook.notes.length) {
        state.selectedNoteId = state.notes[0].id
      }
    },
    updateNote(state, note) {
      let notes = resolveNotes(state)
      let found = notes.find((n) => n.id === note.id)
      for (let prop in found) {
        found[prop] = note[prop]
      }
    },
    selectNote,
    selectFirst,
    setLoadNotebook(state, id) {
      state.loadNotebookId = id
    },
    addChange(state, { type, payload }) {
      switch (type) {
        case 'add':
        case 'delete':
          state.changes[type] = state.changes[type].concat([payload])
          break
        case 'update':
          const hasChange = payload.id in state.changes.update
          const originalNote = (state.notebook.notes || []).find(
            (n) => n.id === payload.id,
          )

          if (
            originalNote &&
            originalNote.content === payload.content &&
            hasChange
          ) {
            delete state.changes.update[payload.id]
          } else {
            state.changes.update[payload.id] = payload
          }
          break
      }
    },
    removeChange(state, { type, payload }) {
      switch (type) {
        case 'add':
        case 'delete':
          state.changes[type] = state.changes[type].filter(
            (n) => n.id !== payload.id,
          )
          break
        case 'update':
          delete state.changes.update[payload.id]
      }
    },
    revertChanges(state) {
      state.changes.add = []
      state.changes.delete = []

      Object.keys(state.changes.update).forEach((id) => {
        let note = (state.notebook.notes || []).find(
          (n) => n.id === parseInt(id),
        )

        if (!note) return

        let content = note.content
        note.content = ''
        note.content = content
      })

      state.changes.update = {}

      if (!resolveNotes(state).length) {
        selectFirst(state) // select none, i.e. -1
      }
    },
    reset(state) {
      for (let prop in emptyState) {
        state[prop] = JSON.parse(JSON.stringify(emptyState[prop]))
      }
      updateUrl(state)
    },
  },
  actions: {
    listNotebooks({ commit }) {
      return api.list().then((data) => commit('setNotebooks', data))
    },
    loadNotebook({ commit }, { id, password }) {
      return api
        .getNotes(id, password)
        .then((data) => commit('setNotes', data))
        .then(() => commit('setNotebook', { id, password }))
    },
    createNotebook({ commit }, { id, password }) {
      return api
        .create(id, password)
        .then((data) => commit('setNotes', data.notes))
        .then(() => commit('setNotebook', { id, password }))
    },
    updateNotebook({ state, commit }, notebook) {
      let updateNotebook = JSON.parse(JSON.stringify(notebook))
      if (notebook.id === state.notebook.id) delete updateNotebook.id
      return api
        .update(state.notebook.id, state.notebook.password, updateNotebook)
        .then(() => commit('setNotebook', notebook))
    },
    createNote({ state, commit }, note) {
      return api
        .addNote(state.notebook.id, state.notebook.password, note)
        .then((data) =>
          commit('addNote', { ...note, id: data.id, oldId: note.id }),
        )
    },
    updateNote({ state, commit }, note) {
      return api
        .updateNote(state.notebook.id, state.notebook.password, note)
        .then(() => commit('updateNote', note))
    },
    deleteNote({ state, commit }, note) {
      return api
        .deleteNote(state.notebook.id, state.notebook.password, note)
        .then(() => commit('removeNote', note))
    },
    applyChanges({ state, commit, dispatch }) {
      let promises = []

      Object.values(state.changes.update).forEach((n) => {
        let newNote = state.changes.add.find((n1) => n1.id === n.id)
        let deletedNote = state.changes.delete.find((n1) => n1.id === n.id)

        if (newNote) {
          commit('updateNote', n)
          commit('removeChange', { type: 'update', payload: n })
        } else if (deletedNote)
          commit('removeChange', { type: 'update', payload: n })
        else
          promises.push(
            dispatch('updateNote', n).then(() =>
              commit('removeChange', { type: 'update', payload: n }),
            ),
          )
      })

      state.changes.add.forEach((n) => {
        if (state.changes.delete.some((n1) => n1.id === n.id)) {
          commit('removeChange', { type: 'add', payload: n })
          commit('removeChange', { type: 'delete', payload: n })
        } else
          promises.push(
            dispatch('createNote', n).then(() =>
              commit('removeChange', { type: 'add', payload: n }),
            ),
          )
      })
      state.changes.delete.forEach((n) => {
        promises.push(
          dispatch('deleteNote', n).then(() =>
            commit('removeChange', { type: 'delete', payload: n }),
          ),
        )
      })

      return Promise.all(promises)
    },
  },
  strict: true,
})

export function useStore() {
  return store
}
