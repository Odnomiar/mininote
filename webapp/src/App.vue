<template>
  <div id="app" class="flex flex-col w-full h-full">
    <div id="alert-container" class="absolute flex justify-center w-full">
      <div
        v-if="alert"
        class="w-1/3 px-6 py-3 rounded-md"
        :class="{
          'alert-success': alert.variant === 'success',
          'alert-danger': alert.variant === 'danger',
        }"
      >
        {{ alert.text }}
      </div>
    </div>

    <div class="flex flex-col flex-grow">
      <control-bar :has-changes="hasChanges" @alert="showAlert"></control-bar>
      <div v-if="!loaded && notebooks.length" class="flex justify-center mt-48">
        <div class="w-1/3">
          <div>
            <h3>Your Notebooks</h3>
            <notebooks-picker
              :notebooks="notebooks"
              @notebook-selected="onNotebookSelected"
            ></notebooks-picker>
          </div>
        </div>
      </div>
      <div v-if="loaded" class="flex justify-between flex-grow mt-8 mb-4">
        <notes-picker
          @noteSelected="onNoteSelected"
          @alert="showAlert"
          @addNote="addNote"
          @deleteNote="deleteNote"
        ></notes-picker>
        <notes-editor
          v-if="selectedNoteId >= 0"
          :id="selectedNoteId"
          class="flex-grow ml-12"
          :content="currentContent"
          :dirty="dirty"
          @alert="showAlert"
          @content-update="onNoteUpdated"
        ></notes-editor>
      </div>
      <div
        v-if="!loaded && !notebooks.length"
        class="flex justify-center mt-64"
      >
        <span class="text-3xl text-center" style="max-width: 400px"
          >Please create a new notebook.</span
        >
      </div>
    </div>
    <div class="text-xs text-gray-400 footer">
      Version {{ version }} | Made w/ &#x2661; by
      <a href="https://muetsch.io">Ferdinand MÃ¼tsch</a> |
      <a href="https://github.com/muety/mininote">GitHub</a>
    </div>
  </div>
</template>

<script>
  import { mapActions, mapGetters, mapMutations, mapState } from 'vuex'
  import { actions, getters, mutations, state } from '@/store/types'

  import NotesEditor from '@/components/NotesEditor'
  import NotesPicker from '@/components/NotesPicker'
  import NotebooksPicker from '@/components/NotebooksPicker'
  import ControlBar from '@/components/ControlBar'

  export default {
    components: {
      NotesEditor,
      NotesPicker,
      NotebooksPicker,
      ControlBar,
    },
    data() {
      return {
        alert: null,
      }
    },
    computed: {
      ...mapState([state.SELECTED_NOTE_ID, state.VERSION, state.NOTEBOOKS]),
      ...mapGetters([
        getters.LOADED,
        getters.SELECTED_NOTE,
        getters.DIRTY,
        getters.CURRENT_CONTENT,
        getters.NOTE_BY_ID,
      ]),
      hasChanges: function () {
        return false
      },
    },
    mounted() {
      window.addEventListener('beforeunload', this.didIntentToLeave)
      this.listNotebooks()
    },
    methods: {
      ...mapActions([actions.LIST_NOTEBOOKS]),
      ...mapMutations([
        mutations.SELECT_NOTE,
        mutations.SET_LOAD_NOTEBOOK,
        mutations.ADD_CHANGE,
      ]),
      onNoteSelected: function (noteId) {
        this.selectNote(noteId)
      },
      onNotebookSelected: function (notebookId) {
        this.setLoadNotebook(notebookId)
      },
      addNote: function (note) {
        this.notes.push(note)
      },
      deleteNote: function (note) {
        this.notes.splice(this.notes.indexOf(note), 1)
      },
      discardChanges: function () {
        this.notes = JSON.parse(JSON.stringify(this.notesInitial))
      },
      showAlert: function (text, variant) {
        let vm = this
        if (!variant) variant = 'danger'
        vm.alert = { text, variant }
        setTimeout(function () {
          vm.alert = null
        }, 3000)
      },
      onNoteUpdated: function ({ content, id }) {
        this.addChange({
          type: 'update',
          payload: { ...this.noteById(id), content },
        })
      },
      didIntentToLeave: function (event) {
        if (this.hasChanges) {
          // Cancel the event as stated by the standard.
          event.preventDefault()
          // Chrome requires returnValue to be set.
          event.returnValue = ''
        }
      },
    },
  }
</script>

<style>
  .alert-success {
    @apply bg-green-600 text-gray-300 border-green-700;
  }

  .alert-danger {
    @apply bg-red-700 text-gray-300 border-red-800;
  }
</style>
