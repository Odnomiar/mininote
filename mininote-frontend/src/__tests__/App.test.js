import { createLocalVue, shallowMount } from '@vue/test-utils'
import BootstrapVue from 'bootstrap-vue'
import App from '@/App'
import store from '../store'

const localVue = createLocalVue()

localVue.use(BootstrapVue)

describe('App.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(App, {
      localVue,
      propsData: {},
      store
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
