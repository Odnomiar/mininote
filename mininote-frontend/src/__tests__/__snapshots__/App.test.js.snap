// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue matches snapshot 1`] = `
<div id="app">
  <div class="logo"><span>✎ Mini_Note</span> <span class="small">BETA </span></div>
  <div class="container-fluid">
    <!---->
    <!---->
    <controlbar-stub></controlbar-stub>
    <!---->
    <div>
      <div class="placeholder"><span>Please open an existing notebook or create a new one.</span></div>
    </div>
  </div>
  <div class="footer">Made w/ ♡ by
    <a href="http://ferdinand-muetsch.de">Ferdinand Mütsch</a> |
    <a href="http://github.com/n1try/mininote">GitHub</a></div>
</div>
`;
